### PR TITLE
Adding option to exclude cameras from recording

### PR DESCRIPTION
**Pull request by jpaussa**
I've never written js before so open to any suggestions. Wanted an easier option to just exclude some cameras I don't need to exported that are high traffic or private. 

Exclude Filtering:
```
[2024-08-15T23:59:06.568Z] Found cameras: [
  "* : Adult Kid's Room",
  '* : Front Door',
  '* : Rear Patio',
  '* : Workshop Rear Area',
  '* : NE Corner Driveway ',
  '* : G4 Bullet ',
  '* : NE Corner Patio (Front Door)',
  '* : Front Garden 360',
  '* : NW Corner Patio Gate',
  '* : Workshop Backyard',
  "* : Kid's Play Room",
  '* : Workshop Interior',
  '* : Garage Interior',
]
[2024-08-15T23:59:06.585Z] Subscribing to motion events for cameras: [
  'Front Door',
  'Rear Patio',
  'Workshop Rear Area',
  'NE Corner Driveway ',
  'G4 Bullet ',
  'NE Corner Patio (Front Door)',
  'Front Garden 360',
  'NW Corner Patio Gate',
  'Workshop Backyard',
  'Workshop Interior',
  'Garage Interior',
]
```

Include Filtering:
```
[2024-08-16T00:01:11.045Z] Found cameras: [
  "* : Adult Kid's Room",
  '* : Front Door',
  '* : Rear Patio',
  '* : Workshop Rear Area',
  '* : NE Corner Driveway ',
  '* : G4 Bullet ',
  '* : NE Corner Patio (Front Door)',
  '* : Front Garden 360',
  '* : NW Corner Patio Gate',
  '* : Workshop Backyard',
  "* : Kid's Play Room",
  '* : Workshop Interior',
  '* : Garage Interior',
]
[2024-08-16T00:01:11.062Z] Subscribing to motion events for cameras: [ 'Garage Interior' ]
```
Both:
```
[2024-08-16T00:00:42.619Z] CAMERA_NAMES and CAMERA_NAMES_EXCLUDE cannot both be specified.
```

### CODE REVIEW

File: src/controller.ts [hunk 1]
```diff
@@ -54,10 +58,19 @@ export default class Controller {
...
-      ? allCameras.filter((camera) => this.cameraNames.includes(camera.name))
-      : allCameras;
+    let filteredCameras;
+    /** Include Filtering */
+    if (this.cameraNames.length) {
+      filteredCameras = allCameras.filter((camera) => this.cameraNames.includes(camera.name));
+    /** Exclude filtering */
+    } else if (this.cameraNamesExclude.length) {
+      filteredCameras = allCameras.filter((camera) => !this.cameraNamesExclude.includes(camera.name));
+      /** No filttering, include all cameras */
```
Comment (bostaunieux) [inline src/controller.ts]:
```suggestion
      /** No filtering, include all cameras */
```

**Review by bostaunieux — COMMENTED** [review]
Thanks for contributing. This is looking good, especially for someone with limited JS experience. I was considering adding more robust include/exclude logic, such as allowing wildcard or regex matches, but that's likely overkill for what's needed here.

There's a minor typo to fix, but you'll also need to update the controller.test.ts file to account for the new `cameraNamesExclude` parameter. If you're able to make that update and get the build workflow to pass, I can approve.

**Comment by jpaussa** [comment]
Cheers, I hadn't seen the tests fail until I got the build failure. Had the same idea about regex's etc but it is a bit of a rabbit hole as you suggest. I'll update this PR with the test fix. I might try in another PR (as I was just trying to get this one to work) but refactoring the filtering into a different method so it can easily be unit tested.

**Comment by bostaunieux** [comment]
Your changes look good, but there's apparently issues running the codecov task in github actions. I've bumped the version of the various actions since they were out of date, but I'll need to spend a bit more time figuring out how to get PRs from contributors to work.

**Pull request closed without merging**